test/test_repeated_variable.jl

```diff
@@ -0,0 +1,43 @@
+using Test
+using ResumableFunctions
+
+# From issue #86
+
+##
+
+@resumable function singleton()::String
+    @yield "anything"
+end
+
+# 1. must be defined before combined!
+@resumable function _empty(x)::String
+end
+
+@resumable function combined()::String
+    for s in singleton()
+        @yield s # fails "here" because s is determined to be of type Nothing because of the second loop
+    end
+    # 2. must reuse of variable
+    for s in _empty(1) # dummy argument needed to illustrate point 1.
+        @yield s
+    end
+end
+
+collect(combined())
```

Comment:
Maybe we could `@test` the return value of `collect` here so it gets counted as a test?